x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/slack/message/SlackMessage.java

```diff
@@ -26,8 +26,16 @@
     final String from;
     final String[] to;
     final String icon;
-    final String text;
-    final Attachment[] attachments;
+    String text;
+    Attachment[] attachments;
```

Comment:
What is the reason for making these non final? Our objects are typically final. And assigning null to these still counts as making sure they are final. It is very atypical for us to assign setters to objects like this to set these values after construction of the object.

Reply:
I can correct this. There's no reason for them not to be final.